### PR TITLE
add md docs for Consine,MM,Mean,Sum,CMaxTable,RoiPooling,SpatialAvera…

### PR DESCRIPTION
**Pull request by shane-huang**
added 12 layers. 

simple layers: Consine,MM,Mean,Sum
merge layers: CMaxTable
Pooling layers: RoiPooling,SpatialAveragePooling, 
Loss functions: DiceCoefficientCriterion,MultiMarginCriterion,DistKLDivCriterion
Convolution layers: SpatialFullConvolution,SpatialConvolution

### CODE REVIEW

**Comment by shane-huang** [comment]
Major issues: 
- the python examples of RoiPooling and SpatialFullConvolution needs to be fixed because of program throws error, issue #1066 #1063 needs to be fixed. 

Minor issues:

- SpaticalFullConvolution, SpatialConvolution python and scala API parameter list unsync. 
- Mean, Sum missing "squeeze" argument in python API. 
- DiceCoefficientCriterion default argument value is missing in python API

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+## SpatialConvolution ##
+
+**Scala:**
+```scala
+val m = SpatialConvolution[T](nInputPlane,nOutputPlane,kernelW,kernelH,strideW,strideH,padW,padH,nGroup,propagateBack,wRegularizer,bRegularizer,initWeight, initBias, initGradWeight, initGradBias)
+```
```
Comment (jason-dai) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: The line is too long; break it into multiple lines.

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+**Python:**
+```python
+m = SpatialConvolution(n_input_plane,n_output_plane,kernel_w,kernel_h,stride_w=1,stride_h=1,pad_w=0,pad_h=0,n_group=1,propagate_back=True,init_method="default")
+```
+
+SpatialConvolution is a module that applies a 2D convolution over an input image.
+
+The input tensor in `forward(input)` is expected to be
+either a 4D tensor (`batch x nInputPlane x height x width`) or a 3D tensor (`batch x height x width`). The convolution is performed on the last two dimensions.
```
Comment (dding3) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: I think if input is 3D tenor, it should be nInputPlan(other than batch) x height x width

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+**Scala example:**
+```scala
+scala> val input = Tensor[Double](1,2,3,3).randn()
```
Comment (dding3) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: As discussed, use float?

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+scala> val gradOut = Tensor[Double](1,1,2,2).fill(0.2)
+gradOut: com.intel.analytics.bigdl.tensor.Tensor[Double] =
+(1,1,.,.) =
+0.2     0.2
+0.2     0.2
+
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 1x1x2x2]
+
+scala> m.backward(input,gradOut)
+res45: com.intel.analytics.bigdl.tensor.Tensor[Double] =
```
Comment (dding3) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: as discussed, remove backward?

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+```
+**Python:**
+```python
+m = SpatialFullConvolution(n_input_plane,n_output_plane,kw,kh,dw=1,dh=1,pad_w=0,pad_h=0,adj_w=0,adj_h=0,n_group=1,no_bias=False,init_method='default',wRegularizer=None,bRegularizer=None,)
+```
+
+SpatialFullConvolution is a module that applies a 2D full convolution over an input image. 
+
+The input tensor in `forward(input)` is expected to be
+either a 4D tensor (`batch x nInputPlane x height x width`) or a 3D tensor (`batch x height x width`). The convolution is performed on the last two dimensions. `adjW` and `adjH` are used to adjust the size of the output image. The size of output tensor of `forward` will be :
```
Comment (dding3) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialFullConvolution.md]: If input is 3D tensor, the first dimention is nInputPlane

**Comment by dding3** [comment]
Actually I think parameters of the layer is important to help user understand how to use these layers, I think it would be better if we give explanations of parameters in the md.

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md (hunk 3 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: changed in the new pull request #1122

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md (hunk 4 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: changed in the new pull request #1122. BTW, as I understand,  we should at least put forward in the example, backward is not necessary but no need to remove if you added any. And we're free to add anything we think user should know. Differences in styles and contents can be tolerated as long as the doc is clear.

File: docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md (hunk 2 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md]: fixed in the new pull request #1122

**Comment by shane-huang** [comment]
replaced this pull request with #1122

**Pull request closed without merging**